CodeConverter/CSharp/MethodBodyExecutableStatementVisitor.cs

```diff
@@ -765,8 +765,7 @@ public override async Task<SyntaxList<StatementSyntax>> VisitSelectBlock(VBSynta
                 }
 
                 var csBlockStatements = (await ConvertStatementsAsync(block.Statements)).ToList();
-                if (csBlockStatements.LastOrDefault()
-                        ?.IsKind(SyntaxKind.ReturnStatement, SyntaxKind.BreakStatement) != true) {
+                if (!WillLastStatementExit(csBlockStatements.LastOrDefault())) {
```

Comment:
This is an example of where the advice from [contributing.md](https://github.com/icsharpcode/CodeConverter/blob/master/.github/CONTRIBUTING.md#implementation-advice) applies:
> Always try to convert directly between the VB and C# model, avoid converting then post-processing the converted result. This prevents the code getting tangled interdependencies, and means you have the full semantic model available to make an accurate conversion.

So in general, I'd have written the analysis method against the Visual Basic syntax tree. That way if you decide there's an interesting case to add that does need the semantic model, you can easily grab it.

That said, what you've written is actually closer to something that could be a wholly independent rosyln analysis/fixer on a C# tree independent of all else. Interestingly a built-in one already exists, so I do wonder if it could be run as part of the general simplification phase to keep the complexity out of here:
![image](https://user-images.githubusercontent.com/2490482/153257985-62dc2a8c-86bd-408c-a9bc-976ff4eedaf6.png)